pramen/extras/src/main/scala/za/co/absa/pramen/extras/sink/StandardizationSink.scala

```diff
@@ -0,0 +1,298 @@
+  *
+  * {{{
+  *  {
+  *    name = "Standardization sink"
+  *    type = "sink"
+  *    sink = "my_standardization_sink"
+  *
+  *    schedule.type = "daily"
+  *
+  *    # Optional dependencies
+  *    dependencies = [
+  *      {
+  *        tables = [ dependent_table ]
+  *        date.from = "@infoDate"
+  *      }
+  *    ]
+  *
+  *    tables = [
+  *      {
+  *        metastore.table = my_table1
+  *        # or
+  *        # input.path = ...
+  *        # job.metastore.table = my_table1
+  *        #
+  *
+  *        output.raw.base.path = "/datalake/base/path/raw/my_table1"
+  *        output.publish.base.path = "/datalake/base/path/publish/my_table1"
+  *
+  *        # All following settings are OPTIONAL
+  *
+  *        # Info version (default = 1)
+  *        output.info.version = 1
+  *
+  *        # Hive table to create/repair
+  *        output.hive.table = "my_hive_table"
+  *
+  *        # Date range to read the source table for. By default the job information date is used.
+  *        # But you can define an arbitrary expression based on the information date.
+  *        # More: see the section of documentation regarding date expressions, an the list of functions allowed.
+  *        date {
+  *          from = "@infoDate"
+  *          to = "@infoDate"
+  *        }
+  *
+  *        transformations = [
+  *         { col = "col1", expr = "lower(some_string_column)" }
+  *        ],
+  *        filters = [
+  *          "some_numeric_column > 100"
+  *        ]
+  *        columns = [ "col1", "col2", "col2", "some_numeric_column" ]
+  *      }
+  *    ]
+  *  }
+  * }}}
+  *
+  */
+class StandardizationSink(sinkConfig: Config,
+                          standardizationConfig: StandardizationConfig) extends Sink {
+
+  import za.co.absa.pramen.extras.sink.StandardizationSink._
+
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  override val config: Config = sinkConfig
+
+  override def connect(): Unit = {}
+
+  override def close(): Unit = {}
+
+  override def send(df: DataFrame,
+                    tableName: String,
+                    metastore: MetastoreReader,
+                    infoDate: LocalDate,
+                    options: Map[String, String])(implicit spark: SparkSession): Long = {
+    implicit val queryExecutor: QueryExecutor = new QueryExecutorSpark(spark)
+
+    val jobStart = Instant.now()
+
+    val infoVersion = getInfoVersion(options)
+
+    val sourceCount = df.count()
+
+    val dfToWrite = repartitionIfNeeded(df, sourceCount)
+
+    val rawCount = if (options.contains(RAW_BASE_PATH_KEY)) {
+      val rawBasePath = getBasePath(tableName, RAW_BASE_PATH_KEY, options)
+      val outputRawPartitionPath = getPartitionPath(rawBasePath, standardizationConfig.rawPartitionPattern, infoDate, infoVersion)
+      writeToRawFolder(dfToWrite, sourceCount, outputRawPartitionPath)
+
+      val rawCount = spark.read.json(outputRawPartitionPath.toUri.toString).count
```

Comment:
Yes. Should be. Just wanted to do a due diligence. The process requires the number of records in raw to be calculated in raw, to be sure it hasn't changed. It can change for instance if Pramen did 'select count(*) from abc', and then 'select * from abc', but. the data is modified between these steps.

Yes, only JSON, at least for now. Really not sure how this sink is going to be evolved at this point. 😄